src/main/java/com/github/paweladamski/httpclientmock/RuleBuilder.java

```diff
@@ -3,14 +3,17 @@
 import com.github.paweladamski.httpclientmock.action.Action;
 import com.github.paweladamski.httpclientmock.condition.Condition;
 import com.github.paweladamski.httpclientmock.condition.HttpMethodCondition;
+import com.github.paweladamski.httpclientmock.condition.UrlEncodedFormCondition;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 import org.hamcrest.Matcher;
 
 class RuleBuilder {
 
   private final List<Action> actions = new ArrayList<>();
   private final List<Condition> conditions = new ArrayList<>();
+  private UrlEncodedFormCondition formParametersCondition;
```

Comment:
It still needs to check for when the user never calls `withFormParameters()` because he is not interested in checking them. In this use-case, the `matches` method should return true no matter how many form parameters the request has, because the user has not told it to check for any.

I can move this logic into the `UrlEncodedFormConditions` class.

Reply:
Fixed in d9ee25b7c17aa84fc00b36b6d5eeca68c22eae0b.